packages/desktop/src/query.rs

```diff
@@ -56,6 +60,64 @@ impl QueryEngine {
             id: request_id,
             reciever: self.sender.subscribe(),
             phantom: std::marker::PhantomData,
+            queue_slab: None,
+            queue_id: None,
+            thread_handle: None,
+        }
+    }
+
+    pub fn new_query_with_comm<V: DeserializeOwned>(
+        &self,
+        script: &str,
+        webview: &WebView,
+        sender: async_channel::Sender<serde_json::Value>,
+    ) -> Query<V> {
+        let request_id = self.active_requests.slab.borrow_mut().insert(());
+        let queue_id = self.active_queues.slab.borrow_mut().insert(());
+
+        let code = format!(
+            r#"
+            {{
+                if (!window.{QUEUE_NAME}) {{
+                    window.{QUEUE_NAME} = [];
+                }}
+
+                let _request_id = {request_id};
+
+                if (!window.{QUEUE_NAME}[{queue_id}]) {{
+                    window.{QUEUE_NAME}[{queue_id}] = [];
+                }}
+                let _message_queue = window.{QUEUE_NAME}[{queue_id}];
+
+                {script}
+            }}
+            "#
+        );
+
+        if let Err(err) = webview.evaluate_script(&code) {
+            log::warn!("Query error: {err}");
+        }
+
+        let thread_receiver = self.sender.subscribe();
+        let thread_handle: JoinHandle<()> = tokio::spawn(async move {
+            let mut receiver = thread_receiver;
+            loop {
+                if let Ok(result) = receiver.recv().await {
+                    if result.id == request_id {
```

Comment:
This ends up scaling poorly with large numbers of requests (explained more in https://github.com/DioxusLabs/dioxus/pull/1115). This shouldn't be any worse than the git version of use_eval. At this point I would like to get this PR merged, so that we can unblock the liveview router. Just something to think about for the future